content/ui-patterns/forms.mdx

```diff
@@ -0,0 +1,387 @@
+---
+title: Forms
+---
+
+import {Box, Heading, Link} from '@primer/components'
+
+<Box sx={{fontSize: 3}} class="lead" as="p">
+    Forms are used to complete tasks that require data input from the user. For example: creating a new repo, configuring settings, and logging in.
+</Box>
+
+<Box sx={{fontSize: 3}} class="lead" as="p">
+    Primer's form design guidelines aim to minimize the effort and cognitive load required to complete a task that involves a form.
+</Box>
+
+## Form control anatomy
+
+![diagrams labeling the anatomy of a text field and a checkbox field](https://user-images.githubusercontent.com/2313998/168928812-fa5c9ad7-4f92-4e9d-8e6e-62eb4413985c.png)
+
+### Label (required)
+
+Labels should be descriptive, and as short as possible: ideally no more than 3 words. Labels should be written in sentence case. For example: "Repository name".
+
+If you're having trouble keeping label text short, consider using a [caption](#caption) to provide more context.
+
+In rare cases, a label may be hidden when the user has enough context to understand what the input does. When a label is visually hidden, label text must still be provided for assistive technology such as screen readers. For example: a search field with an icon that is placed in the spot a user would expect to find a search field may have a visually hidden label that says "Search".
+
+Placeholder text is never an acceptable substitute for a label because:
+
+- The placeholder text disappears as soon as the input has a value
+- Placeholder text colors are typically too light to meet the minimum color contrast ratio required for accessibility
+- Screen readers do not read placeholder text as a label
+
+### Required field indicator
+
+When a field is required to have a value, it should be visibly marked as required. An individual checkbox or radio button cannot be marked as required.
+
+For more information, see the [required fields](#required-fields) section.
+
+### Input (required)
+
+The input is what the user interacts with to set the value of the form control. You may pre-fill inputs with smart default values, but be careful about making too many assumptions about what a user wants or needs.
+
+### Caption
+
+A caption may be used to provide additional context about the field to help users fill in the correct data or explain how the data will be used. Caption text should be as short as possible.
+
+Caption text may be displayed alongside a validation message, or it may be hidden if it only provides redundant information.
+
+Caption text may be used to augment the [label](#label), but should not be redundant with the label or any other parts of the form control. If the caption feels redundant, try removing it. 
+
+<DoDontContainer>
+  <Do>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/166832266-c34634ec-2ade-4bf6-8401-df0bdc02b324.png" />
```

Comment:
![image](https://user-images.githubusercontent.com/293280/169878988-d9667839-ff30-444a-a88f-e0078bcf42b9.png)

@mperrotti  This "do" example feels quite redundant 🤔